app-modules/report/src/Filament/Widgets/Concerns/InteractsWithPageFilters.php

```diff
@@ -76,4 +78,30 @@ public function getMonthRange(Carbon $startDate, Carbon $endDate): Collection
 
         return $months;
     }
+
+    public function getSelectedSegment(): ?string
+    {
+        return $this->filters['populationSegment'] ?? null;
+    }
+
+    /**
+     * @template TModel of \Illuminate\Database\Eloquent\Model
+     *
+     * @param Builder<TModel> $query
+     * @param string $segmentId
+     *
+     * @return void
+     */
+    public function segmentFilter(Builder $query, string $segmentId): void
+    {
+        if (blank($segmentId)) {
+            return;
+        }
+
+        $query->whereKey(
+            app(TranslateSegmentFilters::class)
+                ->execute($segmentId)
+                ->pluck($query->getModel()->getQualifiedKeyName()),
+        );
```

Comment:
I would modify this to intead simply pass the `$query` to the `applyFilterToQuery` method of `TranslateSegmentFilters`.

It is not scalable to run `pluck` and `whereKey` here. Segments can be huge so you may run out of allowed items within an `in` statement, memory, query length, etc.